tests/test_source_mesh.py

```diff
@@ -68,6 +70,31 @@ def test_vertices(source_mesh):
     remove_files()
 
 
+def test_mesh_generation(source_mesh):
+
+    num_s = 6
+    num_theta = 41
+    num_phi = 51
+
+    tets_per_wedge = 3
+    tets_per_hex = 5
+
+    num_elements_exp = tets_per_wedge * (num_theta - 1) * (
+        num_phi - 1
+    ) + tets_per_hex * (num_s - 2) * (num_theta - 1) * (num_phi - 1)
+    num_neg_vols_exp = 0
+
+    remove_files()
+
+    source_mesh.create_vertices()
+    source_mesh.create_mesh()
+
+    assert len(source_mesh.volumes) == num_elements_exp
+    assert len([i for i in source_mesh.volumes if i < 0]) == num_neg_vols_exp
```

Comment:
Nice - does this fail with the old/wrong canonical ordering?

Reply:
With the old canonical ordering, this doesn't fail on the test equilibrium, but it does fail with WISTELL-D. On WISTELL-D using these parameters, the number of elements is as expected but 108 elements (1.47%) have negative volume.